main.cpp

```diff
@@ -270,13 +178,102 @@ struct FloatType
         value = nullptr;
     }
     
+    FloatType& pow(const IntType& x);
+    FloatType& pow(const FloatType& x);
+    FloatType& pow(const DoubleType& x);
+    FloatType& pow(float v);
+    
+    FloatType& powInternal(float x);
+
     FloatType& add( float x );
     FloatType& subtract( float x );
     FloatType& multiply( float x );
     FloatType& divide( float x );
-    operator float() { return  *value; }
+    operator float() const { return  *value; }
 };
 
+/*-------------------------------------------------------------------------------*/
+struct IntType
+{
+private:
+    int* value = nullptr;
+public:
+    IntType( int x ) : value( new int(x) ) { }
+    ~IntType()
+    {   
+        delete value;
+        value = nullptr;
+    }
+
+    IntType& pow(const IntType& x);
+    IntType& pow(const FloatType& x);
+    IntType& pow(const DoubleType& x);
+    IntType& pow(int v);
+    
+    IntType& powInternal(int x);
+    
+    IntType& add( int x );
+    IntType& subtract( int x );
+    IntType& multiply( int x );
+    IntType& divide( int x );
+    operator int() const { return  *value; }
+};
+
+/*-------------------------------------------------------------------------------*/
+struct DoubleType
+{
+private:
+    double* value = nullptr;
+public:
+    DoubleType( double x ) : value( new double(x) ) { }
+    ~DoubleType()
+    {   
+        delete value;
+        value = nullptr;
+    }
+    
+    DoubleType& pow(const IntType& x);
+    DoubleType& pow(const FloatType& x);
+    DoubleType& pow(const DoubleType& x);
+    DoubleType& pow(double v);
+    
+    DoubleType& powInternal(double x);
```

Comment:
This should be `private`.  if it's not private, it's not internal!
- [ ] revise